codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -857,9 +876,15 @@ public Http2Stream closeRemoteSide() {
         }
 
         @Override
-        public int nextStreamId() {
-            // For manually created client-side streams, 1 is reserved for HTTP upgrade, so start at 3.
-            return nextStreamId > 1 ? nextStreamId : nextStreamId + 2;
+        public int incrementAndGetNextStreamId() {
+            return nextReservationStreamId >= 0 ? nextReservationStreamId += 2 : nextReservationStreamId;
```

Comment:
Actually I think we do need this so that we can properly detect that stream creation has been exhausted.  See my comment below ... we could probably use `isExhausted` here too.